### ISSUE TITLE
Home: TypeError: Cannot read property 'indexOf' of null

### ISSUE THREAD

**Issue by jeremybuisson** [platformio/platformio-core#1062]
TypeError: Cannot read property 'indexOf' of null
    at Object.o [as basename] (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:11:49023)
    at t.value (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:31:85404)
    at t.value (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:31:84226)
    at d._renderValidatedComponentWithoutOwnerOrContext (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:31:528221)
    at d._renderValidatedComponent (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:31:528348)
    at d._updateRenderedComponent (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:31:527652)
    at d._performComponentUpdate (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:31:527453)
    at d.updateComponent (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:31:526731)
    at d.receiveComponent (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:31:525848)
    at Object.receiveComponent (http://127.0.0.1:8010/b7b2e59db5b772f69810.min.js:17:30804)

What I did...
1) Get the Code zipfile for Windows x64, unzip it
2) Launch Code, install PlatformIO extension and its dependencies
3) Wait, and reload
4) Wait, and reload
5) Install Adafruit Circuit Playground library
6) New project, select Adafruit Flora, untick "Use default location" ---> crash

**Comment by ivankravets** [comment]
Do you have non-ascii chars in folders/files located in %USER_HOME% directory?

**Comment by jeremybuisson** [comment]
%USER_HOME% contains non-ascii chars, yes.

**Comment by ivankravets** [comment]
Please run this command in PIO IDE Terminal:
```
python -c "import os; print os.listdir(os.path.expanduser('~'))"
```

Do you see non-ascii chars?

**Comment by jeremybuisson** [comment]
Silly question... How do I find PIO IDE Terminal?

**Comment by ivankravets** [comment]
http://docs.platformio.org/en/latest/ide/vscode.html#platformio-toolbar

**Comment by jeremybuisson** [comment]
Would it be the same if I run the command from the C:\.platformio\... location?
If so: yes, there are non-ascii chars in "Voisinage réseau", "Menu Démarrer", "Modèles", etc.
Sorry for the French words with accents...

I'm really sorry if I look dumb, but, in the bottom-left corner, I see only the PIO Home link.
![image](https://user-images.githubusercontent.com/6871902/30378574-f8a23bd2-9893-11e7-9dba-467c6ed8e5fc.png)

**Comment by ivankravets** [comment]
Could you open PIO IDE Terminal and type
```
pio upgrade
pio update
```
Restart IDE.

Does it work now?

**Comment by jeremybuisson** [comment]
I run the commands in the terminal. Now everything is up-to-date with version 3.5.0a15.

But it does not change anything. Now the stack trace is (same error / different file name):
TypeError: Cannot read property 'indexOf' of null at Object.o [as basename] (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:11:40221) at t.value (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:31:86316) at t.value (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:31:85138) at d._renderValidatedComponentWithoutOwnerOrContext (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:31:519352) at d._renderValidatedComponent (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:31:519479) at d._updateRenderedComponent (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:31:518783) at d._performComponentUpdate (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:31:518584) at d.updateComponent (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:31:517862) at d.receiveComponent (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:31:516979) at Object.receiveComponent (http://127.0.0.1:8010/993c9d594c3963b6a512.min.js:17:27043)

**Comment by candidoneto89** [comment]
Having the same issue. I have folders with non-ascii characters too. Tried what was suggested and still facing this problem.

**Comment by ivankravets** [comment]
Thanks for the report!

Please open PIO IDE Terminal and type `pio upgrade`. It should be fixed in 3.5.0b1. Don't forget to restart IDE.
